src/main/kotlin/frc/robot/robotstate/RobotTriggers.kt

```diff
@@ -21,41 +24,64 @@ val isInDeadZone = Trigger {
 
 @LoggedOutput(path = COMMAND_NAME_PREFIX)
 val atShootingRotation =
-    turret.isAtSetpoint.and {
+    Turret.isAtSetpoint.and {
         drive.pose.rotation.measure.isNear(
             swerveCompensationAngle.measure,
             ROTATION_TOLERANCE
         )
     }
 
 val isShooting = Trigger { state == RobotState.SHOOTING }
+val isStaticShooting = Trigger { state == RobotState.STATIC_SHOOTING }
 val isIntaking = Trigger { state == RobotState.INTAKING }
-private val hasFrontBall = roller.hasBall
-val hasBackBall = hopper.hasBall
+private val hasFrontBall = Roller.hasBall
+val hasBackBall = Hopper.hasBall
 private val ballsEmpty = hasFrontBall.or(hasBackBall).negate()
 
 fun bindRobotCommands() {
     isShooting.apply {
-        and(ballsEmpty).onTrue(setIntaking(), stopShooting())
-        and(!isInDeadZone, atShootingRotation).onTrue(startShooting())
+        and(ballsEmpty.and { !forceShoot })
+            .onTrue(setIntaking(), stopShooting())
+        and(!isInDeadZone).apply {
+            and(atShootingRotation, { !ShootOnMove.get() })
+                .onTrue(startShooting())
+            and { ShootOnMove.get() }.onTrue(startShooting())
+        }
```

Comment:
This is a great improvement, I think this is a bit better:

```suggestion
        and(!isInDeadZone).apply {
            val shouldShootOnMove = Trigger(ShootOnMove::get)
            and(atShootingRotation, !shouldShootOnMove)
                .onTrue(startShooting())
            and(shouldShootOnMove).onTrue(startShooting())
        }
```
But don't you also need to check that the hood and flywheel are at their setpoints?
In general, I don't think this binding needs to know whether the shot is compensated or not. It just needs to verify whether the shooter's subsystems are at their setpoints or not, without considering whether they are compensated or not.

Reply:
startShooting check the this : But don't you also need to check that the hood and flywheel are at their setpoints?

Reply:
Okay, it doesn't check the hood though, and it's pretty confusing that the turret is checked here, but hood and flywheel elsewhere... better to combine.

Reply:
c7c11362e7e5293d10df0c2d4614e6d367faaa88